app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -10,25 +10,72 @@ import android.view.ViewGroup
 import android.widget.Toast
 import kotlinx.android.synthetic.main.fragment_login.*
 import kotlinx.android.synthetic.main.fragment_login.view.*
-import org.fossasia.openevent.general.MainActivity
 import org.fossasia.openevent.general.R
 import org.fossasia.openevent.general.utils.Utils
 import org.koin.android.architecture.ext.viewModel
+import com.google.android.gms.common.api.GoogleApiClient
+import com.google.android.gms.auth.api.Auth
+import com.google.android.gms.auth.api.credentials.Credential
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.api.Status
+import android.content.IntentSender
+import org.fossasia.openevent.general.MainActivity
+import com.google.android.gms.common.api.CommonStatusCodes
+import com.google.android.gms.auth.api.credentials.CredentialRequest
+import org.fossasia.openevent.general.utils.nullToEmpty
+import timber.log.Timber
 
-class LoginFragment : Fragment() {
+class LoginFragment : Fragment(), GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener {
 
     private val loginActivityViewModel by viewModel<LoginFragmentViewModel>()
     private lateinit var rootView: View
+    private var googleApiClient: GoogleApiClient? = null
+    private var isResolving: Boolean = false
+    private var isRequesting: Boolean = false
+    private var SAVE_DATA: Int = 1
+    private var FETCH_DATA: Int = 3
+    private lateinit var credentialLogin: Credential
+    private var mode: Int = 0
+
+    override fun onConnected(p0: Bundle?) {
+        //Request credentials from a logged in account
+        Auth.CredentialsApi.disableAutoSignIn(googleApiClient)
+        requestCredentials()
+    }
+
+    override fun onConnectionSuspended(p0: Int) {}
+
+    override fun onConnectionFailed(p0: ConnectionResult) {}
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_login, container, false)
 
+        googleApiClient = context?.let {
```

Comment:
What's the use of context?

Reply:
.Builder requires it

Reply:
so do this 

`if (context != null and activity !=null)`

Reply:
PR Updated